activiti-cloud-service-common/activiti-cloud-services-dbp-rest/src/main/java/org/activiti/cloud/alfresco/argument/resolver/AlfrescoPageArgumentMethodResolver.java

```diff
@@ -58,6 +62,15 @@ public Pageable resolveArgument(
         );
 
         AlfrescoQueryParameters alfrescoQueryParameters = pageParameterParser.parseParameters(webRequest);
+        boolean isMaxItemsExceedingLimit =
+            !alfrescoQueryParameters.getMaxItemsParameter().isSet() ||
+            alfrescoQueryParameters.getMaxItemsParameter().getValue() > MAX_ITEMS_LIMIT;
+
+        if (isMaxItemsExceedingLimit && MAX_ITEMS_LIMIT_ENABLED) {
+            Pageable limitedPageable = PageRequest.of(0, MAX_ITEMS_LIMIT);
+            return new AlfrescoPageRequest(MAX_ITEMS_LIMIT + 1L, MAX_ITEMS_LIMIT, limitedPageable);
```

Comment:
IMO, silently changing the values provided via query param sounds confusing to me and I think people will report bugs for this. Maybe it's better to raise an exception in this case.